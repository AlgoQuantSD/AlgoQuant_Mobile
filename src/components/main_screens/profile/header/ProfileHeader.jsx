import { useAuthenticator } from "@aws-amplify/ui-react-native";
import { Ionicons } from "@expo/vector-icons";
import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";
import { THEME } from "../../../../general_constants/theme/Theme";
import { editNameModalBuilder } from "../../../general_use/modal/helpers/modalFactory";

export default function ProfileHeader(props) {
  // Get the users initials to render in the gray circle
  const { user } = useAuthenticator((context) => [context.user]);
  const userInitials =
    user?.attributes?.given_name[0] + user?.attributes?.family_name[0];

  // Call the modal builder with props, props contains all the functions needed to set info in the modal
  function handlePencilIconPress() {
    editNameModalBuilder(props);
  }

  return (
    <View
      style={styles.profileHeaderContainer}
      testID="profile-header-container"
    >
      <View style={styles.userInitials}>
        <Text style={styles.userInitialsText} testID="user-initials-text">
          {userInitials}
        </Text>
      </View>
      <View style={styles.fullName}>
        <Text
          style={[styles.text, { paddingRight: "1%", paddingLeft: "3%" }]}
          testID="full-name-text"
        >
          {user?.attributes?.given_name} {user?.attributes?.family_name}
        </Text>
        <TouchableOpacity
          hitSlop={{ top: 30, bottom: 30, left: 30, right: 30 }}
          onPress={handlePencilIconPress}
        >
          <Ionicons
            name={THEME.icon.name.editPencil}
            size={THEME.icon.size.small}
            color={THEME.icon.color}
            testID="pencil-icon"
          />
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  profileHeaderContainer: {
    alignItems: "center",
    justifyContent: "center",
    flex: 0.25,
    width: "100%",
    marginTop: "25%",
    borderBottomWidth: 1,
    borderBottomColor: THEME.profileScreen.dividerColor,
  },
  text: {
    fontSize: THEME.text.fontSize.body,
    color: THEME.text.color.primary,
  },
  userInitialsText: {
    fontSize: THEME.text.fontSize.H1,
    color: THEME.text.color.secondary,
  },
  userInitials: {
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: THEME.profileScreen.profilePicBackgroundColor,
    height: 100,
    width: 100,
    borderRadius: 50,
  },
  fullName: {
    alignItems: "center",
    flexDirection: "row",
    marginTop: "2%",
  },
});
