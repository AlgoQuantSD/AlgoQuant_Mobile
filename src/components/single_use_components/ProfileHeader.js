import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { useAuthenticator } from "@aws-amplify/ui-react-native";
import { THEME } from "../../constants/Theme";
import { Ionicons } from "@expo/vector-icons";
import { TouchableOpacity } from "react-native-gesture-handler";

export default function ProfileHeader(props) {
  const {
    isModalVisible,
    setIsModalVisible,
    setModalType,
    setModalTitle,
    setModalInputLabels,
  } = props;
  const { user } = useAuthenticator((context) => [context.user]);
  const userInitials =
    user?.attributes?.given_name[0] + user?.attributes?.family_name[0];

  function handlePencilIconPress() {
    console.log("Handle pencil icon press");
    setModalType("EDIT_NAME");
    setModalTitle("Edit Name");
    setModalInputLabels([
      { label: "First Name", key: "FIRST_NAME_LABEL" },
      { label: "Last Name", key: "LAST_NAME_LABEL" },
    ]);
    setIsModalVisible(!isModalVisible);
  }
  return (
    <View
      style={styles.profileHeaderContainer}
      testID="profile-header-container"
    >
      <View style={styles.userInitials}>
        <Text style={styles.userInitialsText} testID="user-initials-text">
          {userInitials}
        </Text>
      </View>
      <View style={styles.fullName}>
        <Text
          style={[styles.text, { paddingRight: "1%", paddingLeft: "3%" }]}
          testID="full-name-text"
        >
          {user?.attributes?.given_name} {user?.attributes?.family_name}
        </Text>
        <TouchableOpacity
          hitSlop={{ top: 30, bottom: 30, left: 30, right: 30 }}
          onPress={handlePencilIconPress}
        >
          <Ionicons
            name="pencil"
            size={THEME.text.fontSizeBody}
            color={THEME.colors.foreground}
            testID="pencil-icon"
          />
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  profileHeaderContainer: {
    alignItems: "center",
    justifyContent: "center",
    flex: 0.25,
    width: "100%",
    marginTop: "25%",
    borderBottomWidth: "1px",
    borderBottomColor: THEME.profileScreen.dividerColor,
  },
  text: {
    fontSize: THEME.text.fontSizeBody,
    color: THEME.text.color,
  },
  userInitialsText: {
    fontSize: THEME.text.fontSizeH1,
    color: THEME.text.color,
  },
  userInitials: {
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: THEME.profileScreen.profilePicBackgroundColor,
    height: 100,
    width: 100,
    borderRadius: 50,
  },
  fullName: {
    alignItems: "center",
    flexDirection: "row",
    marginTop: "2%",
  },
});
