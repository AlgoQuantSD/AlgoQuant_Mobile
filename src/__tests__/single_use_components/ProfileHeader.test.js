import React from "react";
import ProfileHeader from "../../components/single_use_components/ProfileHeader";
import { render } from "@testing-library/react-native";

jest.mock("@aws-amplify/ui-react-native", () => ({
  useAuthenticator: jest.fn(() => ({
    user: {
      attributes: {
        given_name: "John",
        family_name: "Doe",
      },
    },
  })),
}));

describe("ProfileHeader", () => {
  it("Should render", () => {
    const tree = render(<ProfileHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders the user's initials", () => {
    const { getByTestId } = render(<ProfileHeader />);
    const userInitialsText = getByTestId("user-initials-text");
    expect(userInitialsText.props.children).toEqual("JD");
  });
  it("renders the user's name", () => {
    const { getByTestId } = render(<ProfileHeader />);
    const nameText = getByTestId("full-name-text");
    expect(nameText.props.children).toEqual(["John", " ", "Doe"]);
  });
  it("renders the pencil icon", () => {
    const { getByTestId } = render(<ProfileHeader />);
    const pencilIcon = getByTestId("pencil-icon");
    expect(pencilIcon).toBeTruthy();
  });
});
