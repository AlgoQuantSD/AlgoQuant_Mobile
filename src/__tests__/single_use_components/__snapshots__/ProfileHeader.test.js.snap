// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHeader Should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#747272",
      "borderBottomWidth": "1px",
      "flex": 0.25,
      "justifyContent": "center",
      "marginTop": "25%",
      "width": "100%",
    }
  }
  testID="profile-header-container"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#747272",
        "borderRadius": 50,
        "height": 100,
        "justifyContent": "center",
        "width": 100,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 28,
        }
      }
      testID="user-initials-text"
    >
      JD
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": "2%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 16,
          },
          Object {
            "paddingLeft": "3%",
            "paddingRight": "1%",
          },
        ]
      }
      testID="full-name-text"
    >
      John
       
      Doe
    </Text>
    <RNGestureHandlerButton
      collapsable={false}
      delayLongPress={600}
      enabled={true}
      exclusive={true}
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      hitSlop={
        Object {
          "bottom": 30,
          "left": 30,
          "right": 30,
          "top": 30,
        }
      }
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      touchSoundDisabled={false}
    >
      <View
        accessible={true}
        collapsable={false}
        hitSlop={
          Object {
            "bottom": 30,
            "left": 30,
            "right": 30,
            "top": 30,
          }
        }
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
